src/core/barcodevideofilter.cpp

```diff
@@ -0,0 +1,143 @@
+/***************************************************************************
+ barcodevideofilter.cpp - BarcodeVideoFilter
+
+ ---------------------
+ begin                : 22.07.2022
+ copyright            : (C) 2022 by Mathieu Pellerin
+ email                : mathieu (at) opengis.ch
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "barcodevideofilter.h"
+
+#include <QOpenGLContext>
+#include <QOpenGLFunctions>
+
+class BarcodeDecoderThread : public QThread
+{
+  public:
+    explicit BarcodeDecoderThread( BarcodeDecoder *decoder, const QImage &image )
+      : QThread()
+      , mDecoder( decoder )
+      , mImage( image )
+    {
+    }
+
+  private:
+    void run() override
+    {
+      if ( mDecoder )
+      {
+        mDecoder->decodeImage( mImage );
+      }
```

Comment:
Just wondering how many calls it has per second? If they are too much, we can drain the battery like crazy, probably a small sleep makes sense, so we do 1-2 scans per second max?

Reply:
I've discussed this with @suricactus yesterday. To leave a record of our discussion: capping decoding to say 2 scans per second would probably save some battery (not that much as the camera is what sucks battery out), but it risks having situations where the environment isn't great (shacky hands like mine, shacky barcode, or both) and we'd hit "a bad frame" twice per second and end up having to wait a long longer for the barcode to be decoded.  When decoding as much as possible, we maximize chance of success.

Hence thinking we should go for the decode-as-many-frames-as-possible to begin with.